### PR TITLE
AMQP-474:Log Appenders appId Replacement Once Only

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/AMQP-474

Instead of replacing the placeholder on every log event,
replace it once during initialization.

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j/AmqpAppender.java [hunk 1]
```diff
@@ -527,12 +528,6 @@ public void run() {
 						amqpProps.setMessageId(UUID.randomUUID().toString());
 					}
 
-					// Set applicationId, if we're using one
-					if (null != applicationId) {
-						amqpProps.setAppId(applicationId);
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j/AmqpAppender.java]: Not sure yet that your `replace` solution is good one, but we shouldn't remove AMQP property anyway

Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j/AmqpAppender.java]: doh

**Comment by artembilan** [comment]
@garyrussell , here is a polishing: https://github.com/artembilan/spring-amqp/commit/1e941a8f32f20f0f6da44c3e5ba1bc36c23908e0

**Comment by garyrussell** [comment]
Merged your polish; removed ctor in log4j `EventSender`; squashed,

**Comment by garyrussell** [comment]
Rebased.

**Comment by artembilan** [comment]
Cherry-picked to `1.4.x` as https://github.com/spring-projects/spring-amqp/commit/23bd1dff38f2036141e7d6d6934e5780e9867eaf
To `1.3.x` as https://github.com/spring-projects/spring-amqp/commit/def93af36b5c58be5d28c7aea9c5627b25db2404

**Pull request merged**